openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSPersonServiceImpl.java

```diff
@@ -93,7 +93,20 @@ public Person createPerson(String configName, String givenName, String familyNam
     public Person updatePerson(String configName, Person person) {
         try {
             Config config = configService.getConfigByName(configName);
-            Person updated = personResource.updatePerson(config, person);
+            Person fetchedPerson = personResource.getPersonById(config, person.getUuid());
+
+            Person.Address addressForUpdate = fetchedPerson.getPreferredAddress();
+            if(addressForUpdate != null) {
+                person.getPreferredAddress().setUuid(addressForUpdate.getUuid());
+            }
+            personResource.updatePersonAddress(config, person.getUuid(), person.getPreferredAddress());
+
+            Person.Name nameForUpdate = fetchedPerson.getPreferredName();

```

Comment:
what about causeOfDeath ? I suppose that it also have to be updated separately, however please check it